pkg/activator/net/throttler.go

```diff
@@ -62,15 +62,29 @@ const (
 	revisionMaxConcurrency = queue.MaxBreakerCapacity
 )
 
+func newPodTracker(dest string, b breaker) *podTracker {
+	tracker := &podTracker{
+		dest: dest,
+		b:    b,
+	}
+	tracker.decreaseWeight = func() { tracker.weight.Add(-1) }
+
+	return tracker
+}
+
 type podTracker struct {
 	dest string
 	b    breaker
+
 	// weight is used for LB policy implementations.
 	weight atomic.Int32
+	// decreaseWeight is an allocation optimization for the randomChoice2 policy.
+	decreaseWeight func()
 }
 
-func (p *podTracker) addWeight(w int32) {
-	p.weight.Add(w)
+func (p *podTracker) increaseWeight() func() {
```

Comment:
Im sort of wondering if we actually need the `increaseWeight()` function to return a callback here, because it sort of confused me on first pass why an increaseWeight function is returning a decreaseWeight method. What if this function didn't return anything and randomChoice2 just did `pick.increaseWeight(); return pick.decreaseWeight, pick`?

Reply:
ehr, true! Artifact of playing with different versions I guess.